[GPUP] WebContent process consumes a sandbox extension to the media cache directory unnecessarily

pvollan opened the pull request:
#### 0c06435111bb6f56f6cee07d733519e69ff84b6b
<pre>
[GPUP] WebContent process consumes a sandbox extension to the media cache directory unnecessarily
<a href="https://bugs.webkit.org/show_bug.cgi?id=265406">https://bugs.webkit.org/show_bug.cgi?id=265406</a>
<a href="https://rdar.apple.com/110540943">rdar://110540943</a>

Reviewed by Brent Fulgham.

Disable this code when the GPU process is enabled. As a follow-up, I plan to disable the runtime setting
for Media in the GPU process for Cocoa platforms, since that should never be turned off.

* Source/WebKit/Shared/WebProcessDataStoreParameters.h:
* Source/WebKit/Shared/WebProcessDataStoreParameters.serialization.in:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::webProcessDataStoreParameters):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformSetWebsiteDataStoreParameters):

Canonical link: <a href="https://commits.webkit.org/271677@main">https://commits.webkit.org/271677@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f4dae829a87ddb7db78478434096231ba47ff38b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29212 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7883 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30552 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31801 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26569 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10024 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5188 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26573 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29485 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6543 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25034 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5641 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5783 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33139 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26680 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26505 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32009 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5749 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3931 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29794 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7430 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6262 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3756 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6274 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (commented):
It seems when `ENABLE(GPU_PROCESS)`, web process could still be used for media if `UseGPUProcessForMediaEnabled` is false,  so we cannot remove extension from web process in that case.

brentfulgham commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -6723,7 +6723,7 @@ UseGPUProcessForMediaEnabled:
   humanReadableDescription: "Do all media loading and playback in the GPU Process"
   webcoreBinding: none
-  condition: ENABLE(GPU_PROCESS) && !USE(GSTREAMER)
+  condition: ENABLE(GPU_PROCESS) && !USE(GSTREAMER) && !PLATFORM(COCOA)
```
I don't understand this change. Don't we do media playback in the GPU Process on Cocoa? I know loading is handled in WebContent, but I'm surprised to see this flag be turned off.

pvollan replied:
That is a good point. The intention behind this change was not to change any behavior, but to make it impossible to toggle this setting off on Cocoa platforms. But this is perhaps not the right way to do that?

Thanks for reviewing!

brentfulgham reviewed (approved):
Looks perfect. r=me!

pvollan committed:
[GPUP] WebContent process consumes a sandbox extension to the media cache directory unnecessarily
https://bugs.webkit.org/show_bug.cgi?id=265406
rdar://110540943

Reviewed by Brent Fulgham.

Disable this code when the GPU process is enabled. As a follow-up, I plan to disable the runtime setting
for Media in the GPU process for Cocoa platforms, since that should never be turned off.

* Source/WebKit/Shared/WebProcessDataStoreParameters.h:
* Source/WebKit/Shared/WebProcessDataStoreParameters.serialization.in:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::webProcessDataStoreParameters):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformSetWebsiteDataStoreParameters):

Canonical link: https://commits.webkit.org/271677@main
